### PR TITLE
Improve macOS CI

### PR DESCRIPTION
**Pull request by Ironicbay**

### CODE REVIEW

File: .azure-pipelines.yml [hunk 1]
```diff
@@ -253,22 +253,26 @@ jobs:
...
-  #     set -eux
-  #     py.test $(pytest.base_args) \
-  #       tests/backend --postgresql --runslow \
-  #       -n auto --max-worker-restart=0 -x \
-  #       --junitxml=test-results-postgresql.xml
-  #   displayName: 'Tests postgresql'
+  - bash: |
+      set -eux
+      py.test $(pytest.base_args) \
+        tests --runmountpoint --runslow \
```
Comment (touilleMan) [inline .azure-pipelines.yml:259]: `--runmountpoint` enable mountpoint related tests, but this is already tested at line 265... is it expected ?

File: .azure-pipelines.yml [hunk 2]
```diff
@@ -253,22 +253,26 @@ jobs:
...
   - bash: |
       set -eux
       py.test $(pytest.base_args) \
         tests --runmountpoint --runslow -m mountpoint -x \
         --junitxml=test-results-mountpoint.xml
     displayName: 'Tests mountpoint'
+  - bash: |
+      set -eux
+      py.test $(pytest.base_args) \
+        tests/backend tests/test_cli.py --postgresql --runslow \
```
Comment (touilleMan) [inline .azure-pipelines.yml:272]: why testing also `tests/test_cli.py` with postgresql enabled ? (we normally only tests `tests/backend` with postgresql)

File: tests/test_scripts.py [hunk 3]
```diff
@@ -69,7 +69,10 @@ def run_testenv():
 
 
 @pytest.mark.slow
-@pytest.mark.skipif(sys.platform == "win32", reason="causes a freeze in appveyor for some reasons")
+@pytest.mark.skipif(
+    "linux" not in sys.platform,
+    reason="causes a freeze in appveyor for some reasons, and crashes on Catalina",
```
Comment (touilleMan) [inline tests/test_scripts.py:74]: ah !

File: .azure-pipelines.yml (hunk 1 above)
Comment (Ironicbay) [reply .azure-pipelines.yml:259]: I couldn't figure if there was a difference with the `-m mountpoint` option in line 266. If it's redundant I will remove the mountpoint tests then

File: .azure-pipelines.yml (hunk 2 above)
Comment (Ironicbay) [reply .azure-pipelines.yml:272]: I copied the lines from the linux tests (180th line). Should we fix it on both ?

File: tests/test_scripts.py (hunk 3 above)
Comment (Ironicbay) [reply tests/test_scripts.py:74]: It works fine on Big Sur though, there has always been issues with the test_env on Catalina. I guess the `skipif` will remain until we can run tests on a 11.x image

File: .azure-pipelines.yml (hunk 1 above)
Comment (Ironicbay) [reply .azure-pipelines.yml:259]: Nevermind, I get it now

File: .azure-pipelines.yml (hunk 2 above)
Comment (touilleMan) [reply .azure-pipelines.yml:266]: @vxgmichel @Max-7 Any idea why we enable postgresql in `test_cli.py` ?

**Review by touilleMan — APPROVED** [review]
all good ^^

**Pull request merged**